cmd/openshift-install/targets.go

```diff
@@ -68,18 +68,17 @@ func newTargetsCmd() []*cobra.Command {
 
 func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		assetStore := &asset.StoreImpl{}
-		err := assetStore.Load(rootOpts.dir)
+		assetStore, err := asset.NewStore(rootOpts.dir)
 		if err != nil {
-			logrus.Errorf("Could not load assets from state file: %v", err)
+			return errors.Wrapf(err, "failed to create asset store")
 		}
+
 		for _, a := range targets {
-			err := assetStore.Fetch(a)
-			if err != nil {
+			if err := assetStore.Fetch(a); err != nil {
 				if exitError, ok := errors.Cause(err).(*exec.ExitError); ok && len(exitError.Stderr) > 0 {
 					logrus.Error(strings.Trim(string(exitError.Stderr), "\n"))
 				}
-				err = errors.Wrapf(err, "failed to generate %s", a.Name())
+				return errors.Wrapf(err, "failed to fetch %s", a.Name())
```

Comment:
Again this is regressing the previous behavior that we want to try to persist files from asset if there was some partial success, example try to write `metadata.json` from `Cluster` ....